### PR TITLE
Add unit test for ProcessFile

### PR DESCRIPTION
**Pull request by shihanng**
Hi, this is an attempt to add unit test for ProcessFile which I think it is not covered in the generator pacakage test suite? Hope you find this useful. Thank you.

### CODE REVIEW

**Comment by ekhabarov** [comment]
@shihanng thanks again for the PR. TIL about `.golden` files. Great idea! I'll review and merge it a bit later.

**Comment by shihanng** [comment]
@ekhabarov, thank you. It's my first time using ginkgo testing framework. Please let me know if there is anything I can do to improve this PR. ❤️

File: generator/file_test.go [hunk 1]
```diff
@@ -124,6 +125,52 @@ var _ = Describe("File", func() {
...
+				f = &descriptor.FileDescriptorProto{
+					Options: &descriptor.FileOptions{},
+					Name:    sp("product.proto"),
+					Package: sp("pb"),
+					MessageType: []*descriptor.DescriptorProto{
+						{
+							Name: sp("Product"),
+							Field: []*descriptor.FieldDescriptorProto{
+								&descriptor.FieldDescriptorProto{
+									Name:     sp("ID"),
```
Comment (ekhabarov) [inline generator/file_test.go]: Could you, please:
* Rebase master onto your branch
* Rename this field to `id` (lower case). For details see #14 

Thanks!

Comment (shihanng) [reply generator/file_test.go]: Thanks for the feedback. I've addressed the above.

**Pull request merged**